### PR TITLE
Fix/37847 - Show the tags in the first policyTagLists

### PR DESCRIPTION
**Pull request by Tony-MK**
### Fixed Issues

$ https://github.com/Expensify/App/issues/37847
$ https://github.com/Expensify/App/issues/38683
PROPOSAL: https://github.com/Expensify/App/issues/37847#issuecomment-1982018079

### Tests

**Prerequisite**

On Old Dot, enable multiple levels of tags. Then create a CSV of tags so that:

It includes multiple levels of tags
In the first tag list (or first level of tags), add a special tag name that has the pattern "Parent: Child"

and import that CSV

1. Go to Collect Workspace.
2. Tap the Tags on the menu.
3. Verify it only displays tag names of the first tag list
4. Verify it displays as "Parent: Child" for the special tag name above
5. Tap on that special tag name to open the Tag Settings page
6. Verify it displays as "Parent: Child"
7. Click to edit tag name
8. Verify it displays as "Parent: Child"

- [x] Verify that no errors appear in the JS console

### QA Steps

1. Go to Collect Workspace.
2. Tap the Tags on the menu.
3. Verify it only displays tag names of the first tag list
4. Verify it displays as "Parent: Child" for the special tag name above
5. Tap on that special tag name to open the Tag Settings page
6. Verify it displays as "Parent: Child"
7. Click to edit tag name
8. Verify it displays as "Parent: Child"

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

Android - Native

</details>

<details>
<summary>Android: mWeb Chrome</summary>

Android - mWeb

</details>

<details>
<summary>iOS: Native</summary>

ios - Native

</details>

<details>
<summary>iOS: mWeb Safari</summary>

ios - mWeb

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/22982173/09f486a3-91b5-4c46-8c4f-cc46df410366

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/22982173/bbac2935-adea-47c1-897f-78b058586a14

</details>

### CODE REVIEW

**Review by luacmartins — DISMISSED** [review]

**Comment by allroundexperts** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Comment by allroundexperts** [comment]
@Tony-MK Can you please add test steps?

**Comment by allroundexperts** [comment]
@Tony-MK I can still see different tag lists being displayed even after this change.

**Comment by Tony-MK** [comment]
> @Tony-MK I can still see different tag lists being displayed even after this change.

Could you show me a screenshot?

**Comment by luacmartins** [comment]
@Tony-MK @allroundexperts any updates here?

**Comment by luacmartins** [comment]
Added https://github.com/Expensify/App/issues/38683 to the list of fixed issues here

**Comment by Tony-MK** [comment]
> > @Tony-MK I can still see different tag lists being displayed even after this change.
> 
> Could you show me a screenshot?

@allroundexperts, Could you show me an example of this happening?

I tried checking again but it worked fine.

**Comment by Tony-MK** [comment]
> @Tony-MK @allroundexperts any updates here?

Waiting for @allroundexperts response's.

**Comment by allroundexperts** [comment]
```
{
    "Region": {
        "name": "Region",
        "orderWeight": 1,
        "tags": {
            "North+test": {
                "enabled": true,
                "name": "North+test"
            },
            "South": {
                "enabled": true,
                "name": "South"
            }
        }
    },
    "State": {
        "name": "State",
        "orderWeight": 0,
        "required": true,
        "tags": {
            "California": {
                "enabled": true,
                "name": "California"
            },
            "North+test": {
                "enabled": false,
                "name": "North+test"
            }
        }
    }
}
```

Screenshot 2024-03-26 at 5 22 13 PM

@Tony-MK As you can see from the data stored in onyx, we want to show only one of the Taglist. Currently,  we do not support multi tag lists.

**Comment by luacmartins** [comment]
@Tony-MK did the comment above help?

**Comment by Tony-MK** [comment]
> @Tony-MK As you can see from the data stored in onyx, we want to show only one of the Taglist. Currently, we do not support multi tag lists.

Hey @luacmartins and @allroundexperts,

According to the [bug description](https://github.com/Expensify/App/issues/37847#issue-2172596102), https://github.com/Expensify/App/issues/37847#issuecomment-2004770860 and https://github.com/Expensify/App/issues/37847#issuecomment-1982183268, I think we are supposed to hide the `Parent: Child` tags and display the root level for now.

Right?

**Comment by luacmartins** [comment]
Yes, that's correct!

**Comment by luacmartins** [comment]
@Tony-MK we also have conflicts

**Comment by Tony-MK** [comment]
Hey @luacmartins and @allroundexperts, solve the merge conflicts.

It's ready to be merged.

**Comment by luacmartins** [comment]
@Tony-MK conflicts again. Bump @allroundexperts for review

**Review by luacmartins — DISMISSED** [review]

File: src/pages/workspace/tags/WorkspaceTagsPage.tsx [hunk 1]
```diff
@@ -84,7 +84,11 @@ function WorkspaceTagsPage({policyTags, route}: WorkspaceTagsPageProps) {
         () =>
             policyTagLists
                 .map((policyTagList) =>
-                    lodashSortBy(Object.values(policyTagList.tags || []), 'name', localeCompare).map((value) => {
+                    lodashSortBy(
+                        Object.values(policyTagList.tags || []).filter((value) => PolicyUtils.getCleanedTagName(value.name) === value.name),
```
Comment (Tony-MK) [inline src/pages/workspace/tags/WorkspaceTagsPage.tsx:88]: Should I add a comment since this is a temporary fix?

**Comment by Tony-MK** [comment]
Gentle bump to @allroundexperts for review?

**Comment by luacmartins** [comment]
Reassigning as per https://expensify.slack.com/archives/C02NK2DQWUX/p1711707769397279

**Comment by luacmartins** [comment]
@JmillsExpensify @trjExpensify which one is the correct behavior for simplified collect? https://github.com/Expensify/App/pull/38759#issuecomment-2027390286

**Comment by luacmartins** [comment]
Waiting on @JmillsExpensify or @trjExpensify for confirmation

**Comment by trjExpensify** [comment]
>Tag1
>Tag2: ChildTag2

To confirm, are "Tag 1" and "Tag 2" different levels of tags (i.e customers and departments), or different values in the one level of tags (i.e customer 1, customer 2)?

**Comment by trjExpensify** [comment]
Okay, so that's showing just the one single level of tags. That isn't multi-level tags or dependent tags, so why wouldn't those show all of the single level tag values in the NewDot tags page? So like this:

>Tag1
>Tag2: ChildTag2

Multiple-levels of tags looks like this: 

https://github.com/Expensify/App/assets/16232057/ef7bb195-a47d-4fc1-a4ba-97e80276da28

**Comment by luacmartins** [comment]
@trjExpensify thanks for clarifying that. From your OldDot example of multi-level tags, what should be shown to the user? Right now we only support a single tagList, so we'd only show the tags in the `Projects` list, I believe. Is that right?

**Comment by trjExpensify** [comment]
>so we'd only show the tags in the Projects list, I believe. Is that right?

Correct, we're only supporting one level until the QBO project supports multiple levels. :)

**Comment by luacmartins** [comment]
@Tony-MK cool, so based on Tom's previous comment and the video above, we'd show only the tags in the `Projects` tagList seen at 0:02 in the video, i.e. `000 - Default, Description, J&P Cycles`

**Comment by luacmartins** [comment]
Yes

**Review by luacmartins — COMMENTED** [review]
It seems like we're aligned on the behavior. @Tony-MK what's the ETA to address those changes?

**Comment by Tony-MK** [comment]
Working on it now...

**Comment by luacmartins** [comment]
Just the first tagList

**Comment by trjExpensify** [comment]
@Tony-MK will you be able to complete the changes today? Thanks!

**Comment by Tony-MK** [comment]
Hey @trjExpensify, @luacmartins and @hoangzinh,

Do you think the changes shown below are okay?

Screenshot 2024-04-05 at 02 24 46

**Comment by Tony-MK** [comment]
Here is the screenshot.

Screenshot 2024-04-05 at 04 35 34

**Comment by luacmartins** [comment]
Yea, that seems correct, but as @hoangzinh mentioned, me need to clean up the backslash

**Review by luacmartins — COMMENTED** [review]
LGTM but we need to clean up the backslash in the name

**Review by luacmartins — DISMISSED** [review]

**Comment by luacmartins** [comment]
@hoangzinh please prioritize reviewing this one when you're online 😄

**Comment by luacmartins** [comment]
Yea, we need to fix that

File: src/pages/workspace/tags/WorkspaceTagsPage.tsx [hunk 2]
```diff
@@ -82,17 +82,17 @@ function WorkspaceTagsPage({policyTags, route}: WorkspaceTagsPageProps) {
             fetchTags();
         }, [fetchTags]),
     );
-    const policyTagLists = useMemo(() => PolicyUtils.getTagLists(policyTags), [policyTags]);
+    const policyTagLists = useMemo(() => PolicyUtils.getTagLists(policyTags).slice(0, 1), [policyTags]);
```
[the remark this answers is not in the document]
Comment (Tony-MK) [reply src/pages/workspace/tags/WorkspaceTagsPage.tsx]: Sure

**Comment by luacmartins** [comment]
We can wait until Monday

**Comment by luacmartins** [comment]
@Tony-MK did you get a chance to look at [this issue](https://github.com/Expensify/App/pull/38759#issuecomment-2040913377)? @hoangzinh are you available to finish the review on this one? Let's get this one merged today!

File: src/pages/workspace/tags/WorkspaceTagsPage.tsx [hunk 3]
```diff
@@ -82,29 +81,26 @@ function WorkspaceTagsPage({policyTags, route}: WorkspaceTagsPageProps) {
...
-                            errors: tag.errors ?? undefined,
-                            enabled: tag.enabled,
-                            isDisabled,
-                            rightElement: <RightElementEnabledStatus enabled={tag.enabled} />,
-                        };
-                    }),
-                )
-                .flat(),
-        [policyTagLists, selectedTags],
+            lodashSortBy(policyTagList.tags).map((value) => {
```
Comment (luacmartins) [inline src/pages/workspace/tags/WorkspaceTagsPage.tsx:88]: Why did we change this logic?

**Comment by Tony-MK** [comment]
Working on it...

File: src/pages/workspace/tags/WorkspaceTagsPage.tsx [hunk 4]
```diff
@@ -82,29 +81,26 @@ function WorkspaceTagsPage({policyTags, route}: WorkspaceTagsPageProps) {
             fetchTags();
         }, [fetchTags]),
     );
-    const policyTagLists = useMemo(() => PolicyUtils.getTagLists(policyTags), [policyTags]);
+    // We currently don't support multi level tags, so let's only get the first level tags.
+    const policyTagList = useMemo(() => PolicyUtils.getTagLists(policyTags)[0], [policyTags]);
     const tagList = useMemo<PolicyForList[]>(
         () =>
-            policyTagLists
-                .map((policyTagList) =>
```
[the remark this answers is not in the document]
Comment (Tony-MK) [reply src/pages/workspace/tags/WorkspaceTagsPage.tsx:89]: Sure 👍

**Comment by luacmartins** [comment]
@Tony-MK lint is failing

**Comment by luacmartins** [comment]
We should fix it in this PR @Tony-MK

File: src/pages/workspace/tags/WorkspaceTagsPage.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (Tony-MK) [reply src/pages/workspace/tags/WorkspaceTagsPage.tsx:89]: @hoangzinh, I think we need to slice the `policyTagLists` first.

What do you think?

**Comment by luacmartins** [comment]
@Tony-MK please ping me once this is ready for review again. Also we have conflicts now.

File: src/pages/workspace/tags/WorkspaceTagsPage.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (Tony-MK) [reply src/pages/workspace/tags/WorkspaceTagsPage.tsx]: All Good, thanks.

**Comment by Tony-MK** [comment]
Hey @luacmartins and @hoangzinh, sorry for the delays.

Could we get this PR merged soon?

Thanks.

**Comment by Tony-MK** [comment]
Do we need to change the edit page, as well?

Screenshot 2024-04-10 at 05 40 07

**Comment by luacmartins** [comment]
@Tony-MK let's get this one through the finish line!

**Comment by Tony-MK** [comment]
Finishing today...

**Comment by luacmartins** [comment]
Cool, @Tony-MK please ping @hoangzinh and I once it's ready.

**Comment by Tony-MK** [comment]
@hoangzinh, ready for review?

**Review by luacmartins — DISMISSED** [review]
LGTM. Waiting on @hoangzinh's tests.

**Comment by trjExpensify** [comment]
How we looking here? Can we get this approved and merged today?

**Comment by luacmartins** [comment]
@hoangzinh I think the PR might be ready for review again.

**Comment by luacmartins** [comment]
@Tony-MK I still see [this happening](https://github.com/Expensify/App/pull/38759#issuecomment-2020301599). This is the view in OldDot
Screenshot 2024-04-15 at 8 50 00 AM

And this is what I see on your branch:
Screenshot 2024-04-15 at 8 50 10 AM

This is the data we have in Onyx:
Screenshot 2024-04-15 at 8 52 14 AM

From the above, we need to display only the tags in `policyTags.State.tags`.

So the logic is something like: find the tag list with `orderWeight = 0` render the tags in that list. So for example we should only be showing `California` and `Oregon` to the user based on the example, since those are the tags in the tagList with orderWeight = 0

Screenshot 2024-04-15 at 8 52 14 AM

@Tony-MK can you please update the PR to reflect that?

**Review by luacmartins — CHANGES_REQUESTED** [review]
Left a comment, but I'm still seeing multiple tagLists in NewDot.

**Comment by Tony-MK** [comment]
I think we can review the PR again.

**Comment by Tony-MK** [comment]
The PR is ready for review.

File: src/pages/workspace/tags/WorkspaceEditTagsPage.tsx [hunk 5]
```diff
@@ -81,7 +83,7 @@ function WorkspaceEditTagsPage({route, policyTags}: WorkspaceEditTagsPageProps)
                                     inputID={INPUT_IDS.POLICY_TAGS_NAME}
                                     label={translate(`workspace.tags.customTagName`)}
                                     accessibilityLabel={translate(`workspace.tags.customTagName`)}
-                                    defaultValue={taglistName}
+                                    defaultValue={PolicyUtils.getCleanedTagName(taglistName)}
```
[the remark this answers is not in the document]
Comment (luacmartins) [reply src/pages/workspace/tags/WorkspaceEditTagsPage.tsx:86]: I think this is the tagList, so it might be ok to skip it

**Review by luacmartins — DISMISSED** [review]
LGTM and tested well

**Comment by luacmartins** [comment]
All yours @hoangzinh

**Comment by trjExpensify** [comment]
Ruh-roh, conflicts again @Tony-MK! 😅

**Comment by trjExpensify** [comment]
@Tony-MK can you address that comment and the conflicts, so we can get this one merged? Thanks!

**Comment by Tony-MK** [comment]
I think it is ready.

**Comment by Tony-MK** [comment]
I resolved the comments and conflicts.

File: src/pages/workspace/tags/WorkspaceEditTagsPage.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (luacmartins) [reply src/pages/workspace/tags/WorkspaceEditTagsPage.tsx:86]: I think we can keep it as is and removed the function if we have any issues. I don't think we will though

**Review by luacmartins — APPROVED** [review]

**Comment by luacmartins** [comment]
@hoangzinh I think it's fine to keep that for now. Are you available to complete review?

**Review by hoangzinh — APPROVED** [review]

**Review by luacmartins — APPROVED** [review]

**Pull request merged**